### ISSUE TITLE
32bit and 64bit vulkan mixup in s-r-t-info

### ISSUE THREAD

**Issue by Kuro-Maii** [ValveSoftware/steam-for-linux#7299]
#### Your system information

* Steam client version (build number or date): 0.20200720.0
* Distribution (e.g. Ubuntu): Pop! OS 20.04 LTS
* Opted into Steam client beta?: problem exists in both non-beta as in the beta
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Several games seem to run sub-optimal and some windows games run via proton fail to render anything other then the UI (E.G. Taur) or the game is really dark unless ingame items are actively highlighted  (E.G. VRChat (desktop mode) having an active mirror in view also allows correct rendering)

Digging into the issue steam system information shows that "wrong ELFCLASS" errors for both 32 and 64bit vulkan libraries.

I expected the correct path for 32bit and for 64bit versions of vulkan libraries to be addressed, this clearly does not happen with an out of the box steam installation.
I had the same issue under Debian 10 just last week.
Steam System Information included: [Github Gist](https://gist.github.com/Kuro-Maii/d88a29c9370b7959b987dbacc49ddf85)

#### Steps for reproducing this issue:

1. optional: fresh install of Debian 10.4 (stable) or Pop! OS 20.04
2. install steam
3. open steam client
  - menu: help -> system information

**Comment by kisak-valve** [comment]
Hello @Kuro-Maii, looking at your system information, you have a Sea Islands (CIK) generation video card. The linux kernel uses the `radeon` kernel module by default with this generation of video card. The `radeon` kernel module is incompatible with Vulkan. Please give https://github.com/ValveSoftware/Proton/wiki/For-AMD-users-having-issues-with-non-OpenGL-games a read.

The extended system information tool tries to use both the 32 bit and 64 bit variants of those libraries, one which silently works and the other is expected to harmlessly fail.

**Comment by Kuro-Maii** [comment]
I would like to give it a try, but I am running into the problem that the file `/etc/default/grub` does not exist. 
I ran the following to see if grub would be in a different location, but I don't know if this would be a valid file to edit: 
```
$ sudo find / -iname grub
/usr/lib/recovery-mode/options/grub
```


> The extended system information tool tries to use both the 32 bit and 64 bit variants of those libraries, one which silently works and the other is expected to harmlessly fail.

In my case it isn't one or the other failing, it is both failing.
As the 32 bit lib is addressed by the 64bit system and the 64bit lib being addressed by the 32bit systems.

**Comment by Kuro-Maii** [comment]
After a sleep I searched for [`systemd-boot amdgpu.cik_support=1`](https://duckduckgo.com/?q=systemd-boot+amdgpu.cik_support%3D1) I found the following [post](https://discuss.getsol.us/d/1779-how-to-boot-latest-kernel-after-updating/7) giving me enough hints to find the file I needed to edit.

```
$ sudo -s
# cat /boot/efi/loader/loader.conf 
default Pop_OS-current
#
```
This made me dig around a bit more until I found: `/boot/efi/loader/entries/Pop_OS-current.conf`
I amended this file from:
```
title Pop!_OS
linux /EFI/Pop_OS-d0bab141-8c68-4922-b302-d6e51d180f2c/vmlinuz.efi
initrd /EFI/Pop_OS-d0bab141-8c68-4922-b302-d6e51d180f2c/initrd.img
options root=UUID=d0bab141-8c68-4922-b302-d6e51d180f2c ro quiet loglevel=0 systemd.show_status=false splash
```
to
```
title Pop!_OS
linux /EFI/Pop_OS-d0bab141-8c68-4922-b302-d6e51d180f2c/vmlinuz.efi
initrd /EFI/Pop_OS-d0bab141-8c68-4922-b302-d6e51d180f2c/initrd.img
options root=UUID=d0bab141-8c68-4922-b302-d6e51d180f2c ro radeon.cik_support=0 amdgpu.cik_support=1 quiet loglevel=0 systemd.show_status=false splash
```
going to reboot and see if I now have the correct kernel modules loaded. I will edit this post with either failure or success report.

**Comment by smcv** [comment]
> Digging into the issue steam system information shows that "wrong ELFCLASS" errors for both 32 and 64bit vulkan libraries.

This is <https://github.com/KhronosGroup/Vulkan-Loader/issues/262>. It's harmless.

**Comment by kisak-valve** [comment]
Closing as fixed in the 1.2.176 vulkan loader.
